ui/src/lib/toolUtils.ts

```diff
@@ -83,6 +92,28 @@ export const getToolIdentifier = (tool: Tool): string => {
   return `unknown-tool-${Math.random().toString(36).substring(7)}`;
 };
 
+/**
+ * Parse groupKind string to extract apiGroup and kind
+ * @param groupKind - String in format "kind.apiGroup" or just "kind"
+ * @returns Object with apiGroup and kind
+ */
+export const parseGroupKind = (groupKind: string): { apiGroup: string; kind: string } => {
+  // Handle null, undefined, or empty string
+  if (!groupKind || !groupKind.trim()) {
+    return { apiGroup: "kagent.dev", kind: "MCPServer" };
```

Comment:
nit: since you're using "kagent.dev" and "MCPServer" (and `McpServer`) multiple times, perhaps make them constants at the root level? (also, let's stick with either uppercase version (MCPServer) or the other one (McpServer))